core/services/head_tracker.go

```diff
@@ -134,17 +135,26 @@ type HeadTracker struct {
 // NewHeadTracker instantiates a new HeadTracker using the orm to persist new block numbers.
 // Can be passed in an optional sleeper object that will dictate how often
 // it tries to reconnect.
-func NewHeadTracker(store *strpkg.Store, callbacks []strpkg.HeadTrackable, sleepers ...utils.Sleeper) *HeadTracker {
-	var sleeper utils.Sleeper
-	if len(sleepers) > 0 {
-		sleeper = sleepers[0]
-	} else {
-		sleeper = utils.NewBackoffSleeper()
+func NewHeadTracker(store *strpkg.Store, callbacks []strpkg.HeadTrackable, readyCallback func()) *HeadTracker {
+	return NewHeadTrackerWithSleeper(store, callbacks, readyCallback, utils.NewBackoffSleeper())
+}
+
+// NewHeadTrackerWithSleeper instantiates a new HeadTracker with an explicitly set sleeper
+func NewHeadTrackerWithSleeper(
```

Comment:
Why this refactoring?

Reply:
Not totally necessary. Had to add another param to the function and decided to separate the "optional" sleeper param into it's own function. Feels cleaner to me since it's only used in tests.